.circleci/config.yml

```diff
@@ -39,32 +64,99 @@ jobs:
             - packages/cspell/node_modules
             - node_modules
             - ~/.cache
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - repo
+            - .ssh
+            - .cache
+
   run-regular-tests:
     <<: *defaults
     steps:
-      - checkout
-      - run:
-          name: Setup Environment
-          command: |
-            echo 'export PATH=./node_modules/.bin:$PATH' >> $BASH_ENV
-            echo 'export PATH=$HOME/.yarn/bin:$HOME/.config/yarn/global/node_modules/.bin:$PATH' >> $BASH_ENV
-      - restore_cache:
-          key: v-0-0-4-node_modules_dir-{{ checksum "yarn.lock" }}
+      - attach_workspace:
+          at: /home/circleci
+      - setup-environment
       - run:
           name: Run Tests and Collect Coverage
           command: |
-            cd ~/repo
-            nx run-many --target=build --all
-            yarn run test --coverage -w=2
+            cd ~/repo;
+            yarn run build;
+            yarn run test --coverage -w=2;
       - run:
           name: Upload Coverage Reports
           command: codecov --disable=gcov --token=$CODECOV_TOKEN
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - repo
+
+  publish-packages:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - setup-environment
+      - auth-gcloud
+      - run:
+          name: Checking for Existing Packages
+          command: |
+            echo "Configuring registry auth...";
+            touch ~/.npmrc;
+            yarn run artifactregistry-login;
+
+            WORKSPACE_PACKAGES=( $(yarn workspaces info --json | sed '$d' | tail -n +2 | jq -r 'keys | join(" ")') );
+
+            echo "Detecting existing packages...";
+
+            touch /tmp/packages_to_publish;
+            for PACKAGE in ${WORKSPACE_PACKAGES[@]};
+            do
+              export PACKAGE=$PACKAGE
```

Comment:
Curious - why here (and elsewhere below) do you export `PACKAGE` and reference it using `$PACKAGE` when you could presumably just continue referencing it using `PACKAGE`

Reply:
This is because of how `jq` consumes these variables. If the `PACKAGE` variable wasn't exported - jq qouldn't be able to reference it's value within the clauses I supplied

ex
```
# env variable can't be referenced
WORLD="world";
jq --null-input '{"hello": env.WORLD }';
# prints { "hello": null }

# works as expected
export THERE="there";
jq --null-input '{"hello": env.THERE }';
# prints { "hello": "there" }

```